core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -97,77 +146,243 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
                 get_exploration_opportunity_summary_from_model)
             | 'Group by ID' >> beam.GroupBy(lambda m: m.id)
         )
-
-        user_stats_results = (
+        skill_opportunities = (
+            self.pipeline
+            | 'Get all non-deleted skill opportunity models' >> ndb_io.GetModels(
+                opportunity_models.SkillOpportunityModel.get_all(
+                    include_deleted=False))
+            | 'Transform to skill opportunity domain object' >> beam.Map(
+                opportunity_services.
+                get_skill_opportunity_from_model)
+            | 'Group skill opportunity by ID' >> beam.GroupBy(lambda m: m.id)
+        )
+        
+        exp_opportunity_to_submitted_suggestions = (
             {
-                'suggestion': suggestions_grouped_by_target,
+                'suggestion': translation_suggestions_grouped_by_target,
                 'opportunity': exp_opportunities
             }
             | 'Merge models' >> beam.CoGroupByKey()
-            | 'Get rid of key' >> beam.Values()  # pylint: disable=no-value-for-parameter
-            | 'Generate stats' >> beam.ParDo(
-                lambda x: self._generate_stats(
+            | 'Get rid of key submitted objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        exp_opportunity_to_reviewed_submitted_suggestions = (
+            {
+                'suggestion': reviewed_translation_suggestions_grouped_by_target,
+                'opportunity': exp_opportunities
+            }
+            | 'Merge reviewed models' >> beam.CoGroupByKey()
+            | 'Get rid of key of reviewed objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        skill_opportunity_to_submitted_suggestions = (
+            {
+                'suggestion': question_suggestions_grouped_by_target,
+                'opportunity': skill_opportunities
+            }
+            | 'Merge submitted question models' >> beam.CoGroupByKey()
+            | 'Get rid of key of submitted question objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+        skill_opportunity_to_reviewed_suggestions = (
+            {
+                'suggestion': question_reviews_grouped_by_target,
+                'opportunity': skill_opportunities
+            }
+            | 'Merge reviewed question models' >> beam.CoGroupByKey()
+            | 'Get rid of key of reviewed question objects' >> beam.Values()  # pylint: disable=no-value-for-parameter
+        )
+
+        translation_contribution_stats_results = (
```

Comment:
These are PCollections of (key, stats) tuples, right? Can we update the names to reflect that? E.g. `translation_contribution_stats_keys_and_results`. And maybe add a comment as well.